### PR TITLE
feat(logs): specify log group's region for LogRetention

### PR DESCRIPTION
**Pull request by Stacy-D**
Global AWS services follow the convention of auto-creating CloudWatch Logs in `us-east-1`, which
makes it challenging to control the log group retention when the stack is created in other region.

This change adds support for specifying the optional region for LogRetention to be able to set log retention on CloudWatch logs created by global AWS Services in `us-east-1`

Closes https://github.com/aws/aws-cdk/issues/9703

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by nija-at — COMMENTED** [review]
Thanks for submitting this PR 😊

I would like this one to wait until this PR is merged - https://github.com/aws/aws-cdk/pull/9808.

**Comment by Stacy-D** [comment]
That's fine to wait for https://github.com/aws/aws-cdk/pull/9808. We were doing these changes(move and adding the region) in parallel so they resulted in two dependent PRs. I believe the changes from my side after https://github.com/aws/aws-cdk/pull/9808  should involve moving this code to the LogRetention in aws-logs and addressing any other comments that you might have.

**Comment by humanzz** [comment]
Hey @Stacy-D, https://github.com/aws/aws-cdk/pull/9808 has now been merged and you can now progress this :)

**Comment by humanzz** [comment]
Hey @nija-at, I think this is now ready for further review after [rebasing on my changes](https://github.com/aws/aws-cdk/pull/9808).
Appreciate you having a look so we can move forwards with this.

File: packages/@aws-cdk/aws-logs/README.md
Comment (nija-at) [inline packages/@aws-cdk/aws-logs/README.md:35]: This was missed in the previous PR - can we add a separate section in the README dedicated to the `LogRetention` construct? 

I suspect it will look something similar to - https://github.com/aws/aws-cdk/tree/master/packages/@aws-cdk/aws-lambda#singleton-function

File: packages/@aws-cdk/aws-logs/lib/log-retention.ts
Comment (nija-at) [inline packages/@aws-cdk/aws-logs/lib/log-retention.ts:18]:
```suggestion
   * The region where the log group should be created
   * @default - same region as the stack
```

**Review by nija-at — CHANGES_REQUESTED** [review]
Looks mostly good. Just a couple of comments around documentation.

File: packages/@aws-cdk/aws-logs/README.md
Comment (Stacy-D) [reply packages/@aws-cdk/aws-logs/README.md]: Added a new entry

File: packages/@aws-cdk/aws-logs/README.md
Comment (nija-at) [inline packages/@aws-cdk/aws-logs/README.md:46]:
```suggestion
By default, the log group will be created in the same region as the stack. The `logGroupRegion` property can be used to configure
log groups in other regions. This is typically useful when controlling retention for log groups auto-created by global services that
publish their log group to a specific region, such as AWS Chatbot creating a log group in `us-east-1`.
```

**Review by nija-at — COMMENTED** [review]
I've taken the liberty to re-write some sections in the README.

**Review by nija-at — APPROVED** [review]

**Pull request merged**
